trace/trace.go

```diff
@@ -69,63 +132,121 @@ func New(name string, fields ...Field) *Trace {
 // how long it took. The Fields add key value pairs to provide additional details about the trace
 // step.
 func (t *Trace) Step(msg string, fields ...Field) {
-	if t.steps == nil {
+	if t.traceItems == nil {
 		// traces almost always have less than 6 steps, do this to avoid more than a single allocation
-		t.steps = make([]traceStep, 0, 6)
+		t.traceItems = make([]traceItem, 0, 6)
 	}
-	t.steps = append(t.steps, traceStep{stepTime: time.Now(), msg: msg, fields: fields})
+	t.traceItems = append(t.traceItems, traceStep{stepTime: time.Now(), msg: msg, fields: fields})
 }
 
-// Log is used to dump all the steps in the Trace
-func (t *Trace) Log() {
-	// an explicit logging request should dump all the steps out at the higher level
-	t.logWithStepThreshold(0)
+// Nest adds a nested trace with the given message and fields and returns it.
+func (t *Trace) Nest(msg string, fields ...Field) *Trace {
+	newTrace := New(msg, fields...)
```

Comment:
for convenience we might want to handle the `t==nil` case too, i.e. no new parent. Then you can write `GetTraceFromContext(ctx).Nest` without case distinction between nil or non-nil.

Reply:
only requires to skip the append below if I see it correctly.

Reply:
That would be handy. Adding.